test/ctia/test_helpers/core.clj

```diff
@@ -290,59 +307,71 @@
                                                   :role role
                                                   :capabilities caps})))
 
-(defmacro deftest-for-each-fixture [test-name fixture-map & body]
+(defmacro deftest-for-each-fixture-with-app [test-name fixture-map app & body]
+  (assert (simple-symbol? app) (pr-str app))
   `(do
      ~@(for [[name-key fixture-fn] fixture-map]
          `(test/deftest ~(with-meta (symbol (str test-name "-" (name name-key)))
                                     {(keyword name-key) true})
-            (~fixture-fn (fn [] (do ~@body)))))))
+            (~fixture-fn
+              (fn []
+                (let [~app (get-current-app)]
+                  ~@body)))))))
 
-(defn get-http-port []
-  ;; Note: using current-get-in-config-fn here is a hack and should be refactored
-  ;; to function parameter once the initial TK setup is merged into master.
-  ((current-get-in-config-fn) [:ctia :http :port]))
+(defn get-http-port [app]
+  (let [{{:keys [get-in-config]} :ConfigService} (app/service-graph app)]
+    (get-in-config [:ctia :http :port])))
 
-(def get
-  (mthh/with-port-fn get-http-port mthh/get))
+(defn GET [app & args]
+  (apply (mthh/with-port-fn (partial get-http-port app) mthh/get)
+         args))
 
-(def post
-  (mthh/with-port-fn get-http-port mthh/post))
+(defn POST [app & args]
+  (apply (mthh/with-port-fn (partial get-http-port app) mthh/post)
+         args))
 
-(defn post-bulk [examples]
+(defn POST-bulk [app examples]
   (let [{bulk-res :parsed-body}
-        (post "ctia/bulk"
+        (POST app
+              "ctia/bulk"
               :body examples
               :socket-timeout (* 5 60000)
               :headers {"Authorization" "45c1f5e3f05d0"})]
     bulk-res))
 
-(defn post-entity-bulk [example plural x headers]
+(defn POST-entity-bulk [app example plural x headers]
   (let [new-records
         (for [y (range 0 x)]
           (-> example
               (dissoc :id)
               (assoc :revision y)))]
-    (-> (post "ctia/bulk"
+    (-> (POST app
+              "ctia/bulk"
               :body {plural new-records}
               :headers headers
               :socket-timeout (* 5 60000))
         :parsed-body
         plural)))
 
-(def delete
-  (mthh/with-port-fn get-http-port mthh/delete))
+(defn DELETE [app & args]
+  (apply (mthh/with-port-fn (partial get-http-port app) mthh/delete)
+         args))
 
-(def put
-  (mthh/with-port-fn get-http-port mthh/put))
+(defn PUT [app & args]
+  (apply (mthh/with-port-fn (partial get-http-port app) mthh/put)
+         args))
 
-(def patch
-  (mthh/with-port-fn get-http-port mthh/patch))
+(defn PATCH [app & args]
```

Comment:
I think that the signature could be: `[app, path :- s/Str & {:as options}]` - a little stronger and more informative (for tooling etc)